src/pages/waypoint/docs/[[...page]].tsx

```diff
@@ -52,12 +53,23 @@ export async function getStaticProps({ params }) {
     remarkPlugins: [[anchorLinks, { headings }]],
   })
 
+  const breadcrumbLinks = [
```

Comment:
This could possibly be abstracted even further, for example we could generated the two first links based on the product slug (`waypoint`) and `basePath` (mapping eg `docs` > `Docs`, `commands` > `Commands`, etc consistently for all products). But this felt a little early to try to tackle in detail, and felt like this bit of manual specificity probably isn't so harmful anyways.

Reply:
Keeping in mind that this will expand out to other product pages, I think anything we can do to reduce boilerplate would be a win 👍

Reply:
Sounds good - I'll try out the "pretty name" mapping for `basePath` values, which should let us move the `getDocsBreadcrumbs` call into the docs `layout` rather than having to include it on every docs path 👍

Reply:
err wait will actually split into a separate util, as then we can still call it in `getStaticProps` rather than client-side 👍

Reply:
@BRKalow Tried it out, ended up shying away from the mapping, it felt odd, figured it's easier to pass in the desired "pretty" name for the basePath. This means a bit more boilerplate in the docs `[[...page]]` file though (extra arg to pass). I think the best "boilerplate reduction" solution here may be to move the breadcrumb link generation into `docs-page`'s `generateStaticProps`.

With getting things ready for the demo in mind, have further abstracted the `getDocsBreadcrumbs` function, and added a `TODO` comment noting that we may want to move it into `docs-page` in the future. Let me know what you think! 🙇

Reply:
Sounds good!

Reply:
In my PR for making the main nav links internal, I was able to determine the product slug and what I called the subpage (docs, commands, plugins) here: https://github.com/hashicorp/dev-portal/pull/50/files#diff-4943d44a4e7e2fcf35e1eb0eee646918c1247641ca15621f5b0a28544f336456R14-R20. It's safe to have in that area since those are those three subpages are the only ones that use the sidebar, but it's probably something good to abstract soonish! It's worth noting that we will definitely have a context in the future that tracks the current product slug, it'll be updated by the `ProductSwitcher`. Maybe when we first create that context it pulls the current product slug from the router path?

Reply:
☝️Nice, that seems like a good plan! On board with that as a near-future abstraction 💯